libs/editor/libs/editor-common/assets/ZSSRichTextEditor.js

```diff
@@ -1138,45 +1144,76 @@ ZSSEditor.removeImage = function(imageNodeIdentifier) {
  *  @brief Inserts a video tag using the videoURL as source and posterURL as the
  *  image to show while video is loading.
  *
- *  @param videoURL  the url of the video to present
- *  @param posterURL the url of an image to show while the video is loading
- *  @param alt       the alt description when the video is not supported.
+ *  @param videoURL     the url of the video
+ *  @param posterURL    the url of an image to show while the video is loading
+ *  @param videoPressID the VideoPress ID of the video, when applicable
  *
  */
-ZSSEditor.insertVideo = function(videoURL, posterURL, alt) {
-    var html = '<video controls webkit-playsinline poster="' + posterURL+ '"><source src="'
-    + videoURL + '" type="video/mp4"/>' + alt + '</video>';
+ZSSEditor.insertVideo = function(videoURL, posterURL, videopressID) {
+    var html = '<video webkit-playsinline src="' + videoURL + '" onclick="" controls="controls" preload="metadata"';
+
+    if (posterURL != '') {
```

Comment:
> BTW, should we add a default placeholder if `posterURL == '' || posterURL == null`)?

We can't, there's an issue with video tags on Android where the video will stretch to fit the size of the placeholder image, even if the video has different dimensions (first encountered https://github.com/wordpress-mobile/WordPress-Android/commit/66bea0be6243139e40c34d71c7c5d3108d39b9a2). So we only show the poster while loading, and only use a thumbnail specific to that video (if it exists) once the video element has been updated with the remote `src` of the video.

Agreed on null-checking `posterURL` though.